src/webtransport.js

```diff
@@ -316,50 +416,74 @@ class Http3WTSession {
       }
     })
 
-    this.datagrams = {}
-    this.datagrams.readable = new ReadableStream({
-      start: (controller) => {
-        this.incomDatagramController = controller
-      }
-    })
+    /** @type {Array<() => void>} */
     this.writeDatagramRes = []
+    /** @type {Array<() => void>} */
     this.writeDatagramRej = []
+    /** @type {Array<Promise<void>>} */
     this.writeDatagramProm = []
-    this.datagrams.writable = new WritableStream({
-      start: (controller) => {
-        this.outgoDatagramController = controller
-      },
-      write: (chunk, controller) => {
-        if (this.state === 'closed') throw new Error('Session is closed')
-        if (chunk instanceof Uint8Array) {
-          const ret = new Promise((resolve, reject) => {
-            this.writeDatagramRes.push(resolve)
-            this.writeDatagramRej.push(reject)
-          })
-          this.writeDatagramProm.push(ret)
-          // console.log('b4 datagram write', chunk, Date.now())
-          this.objint.writeDatagram(chunk)
-          return ret
-        } else throw new Error('chunk is not of type Uint8Array')
-      },
-      close: (controller) => {
-        // do nothing
-      }
-    })
 
+    this.datagrams = {
+      readable: new ReadableStream({
+        start: (controller) => {
+          this.incomDatagramController = controller
+        }
+      }),
+      writable: new WritableStream({
+        start: (controller) => {
+          this.outgoDatagramController = controller
+        },
+        write: (chunk, controller) => {
+          if (this.state === 'closed') throw new Error('Session is closed')
+          if (chunk instanceof Uint8Array) {
+            /** @type {Promise<void>} */
+            const ret = new Promise((resolve, reject) => {
+              this.writeDatagramRes.push(resolve)
+              this.writeDatagramRej.push(reject)
+            })
+            this.writeDatagramProm.push(ret)
+            // console.log('b4 datagram write', chunk, Date.now())
+            if (this.objint == null) {
+              throw new Error('this.objint is not set')
+            }
+            this.objint.writeDatagram(chunk)
+            return ret
+          } else throw new Error('chunk is not of type Uint8Array')
+        },
+        close: () => {
+          // do nothing
+        }
+      }),
+      maxDatagramSize: 0,
```

Comment:
Are you sure, that the following properties have suitable default values?

Reply:
Yes `maxDatagramSize` can be pulled from c++ side. But until it is implemented by someone  0 is a very bad value, null would be better. Since it would say that the datagram should have 0 size.
The other values such as `incomingMaxAge`, should also be left out, since this logic is not implemented, and have to be implemented on the js side.

Reply:
What about excluding yet not implemented fields from the type definition (commenting for later)? In this way, the compiler will error if the fields are used giving the authors a hint, that the implementation should differ from the browser side.

Reply:
Concerning the watermarks, it is impossible to support these properties also in the future. Since it is not exposed by the normal stream, which we use. So these have to be removed from the interface.

Reply:
> But until it is implemented by someone 0 is a very bad value, null would be better

The spec type is `readonly attribute unsigned long maxDatagramSize`, not `readonly attribute unsigned long maxDatagramSize?` so it cannot be null.

> What about excluding yet not implemented fields from the type definition

I guess this is the key question.  The spec defines all these fields and their types, if we add types that diverge from the spec then we're not following the spec.

I would err on the side of consistency with the spec even though the values may be nonsensical (because one day they will hopefully be sensical), but I'm getting the feeling you'd rather not implement the spec fully?

Reply:
The spec is for a client in a webbrowser. The server side is fully unspecified. (and yes we have a client as well, but I assume that is not the main application). So there are natural divergences. 
Anyway I try to be as close as possible to the client spec,  so that one can write code, that works on the browser and client in the same way.

And yes I would rather implement the spec not fully. 
Or my approach as follows, I needed the webtransport stuff for a project, that why I wrote. I implemented all that I currently needed. The idea is, that if someone needs a feature, that is currently not implemented, I will help to guide a PR to improve the package functionality.
But I also want clear indications (errors), of what is currently not implemented. I hate it, if I have a lib and after hours of searching I find out, that it errors just, because it pretends to have it.

But if time progresses I hope we will get close and closer to full features in client spec, except for some differences. One is that I use normal web streams, the streams of webtransport are close to webstreams but differ slightly. Therefore I can not implement a completely identical interface.

Reply:
> but I assume that is not the main application

For my usecase it's both as client and server in node.  Essentially making [@libp2p/webtransport](https://www.npmjs.com/package/@libp2p/webtransport) work in node as well as Chrome.

> And yes I would rather implement the spec not fully.

Ok, I'll remove the stuff added to satisfy the spec, then comment it out of the types and add a note to the README about where this module diverges from the spec.  That should hopefully cover all of the bases?

Reply:
Mine is secondary also client. (Audio video router, browser as well as a network of multiple node.js).

And yes this should cover everything once it is done.
Now, back to your arm mac  problem

Reply:
I've done the commenting and added a block at the bottom of the readme

Reply:
Perfect, I will now turn on the unit tests..., and see if all platforms are working. You might want to check out the branch about OSX compilation. I try to do the universal binary in cmakelists.txt directly, so far no success, but I do not have a mac and rely on github runner and this takes some time.